tests/test_compare_pytorch.py

```diff
@@ -0,0 +1,227 @@
+import time
+
+from brains import LstmNN, GruNN, ElmanNN
+from brains.i_layer_based_brain import ILayerBasedBrainCfg
+from tests.pytorch_brains import IPytorchBrainCfg, GruPyTorch, LstmPyTorch, ElmanPyTorch
+
+
+@pytest.fixture
+def torch_config() -> IPytorchBrainCfg:
+    return IPytorchBrainCfg(type="GRU_PyTorch", num_layers=3,
+                            hidden_size=8,
+                            use_bias=False)
+
+
+@pytest.fixture
+def numpy_config() -> ILayerBasedBrainCfg:
+    return ILayerBasedBrainCfg(type="GRULayered", hidden_layer_structure=[8, 8, 8], diagonal_hidden_to_hidden=False,
+                               use_bias=False)
+
+
+
+
+class TestPyTorch:
+
+    def test_gru_output(self, numpy_config: ILayerBasedBrainCfg, torch_config: IPytorchBrainCfg):
+        input_size = 28
+        output_size = 8
+
+        number_of_inputs = 10000
+
+        individual_size = GruPyTorch.get_individual_size(input_size, output_size, torch_config, {})
+
+        # Make two copies to avoid possible errors due to PyTorch reusing data from the same memory address
+        individual_pytorch = np.random.randn(individual_size).astype(np.float32)
+        individual_numpy = np.copy(individual_pytorch)
+
+        with torch.no_grad():
+            gru_pytorch = GruPyTorch(input_size, output_size, individual_pytorch, torch_config, {})
+
+        gru_numpy = GruNN(input_size, output_size, individual_numpy, numpy_config, {})
+
+        # Also initialize the values for the hidden and cell states the same
+        hidden_pytorch = np.random.randn(*gru_pytorch.hidden.size()).astype(np.float32)
+
+        hidden_numpy = np.copy(hidden_pytorch).reshape(len(numpy_config.hidden_layer_structure), -1)
+
+        gru_pytorch.hidden = torch.from_numpy(hidden_pytorch)
+
+        gru_numpy.hidden = [[np.copy(hidden_numpy)[i]] for i in
+                            range(len(np.copy(hidden_numpy)))]
+        with torch.no_grad():
+            for i in range(len(numpy_config.hidden_layer_structure)):
+                attr_weight_ih_li = "weight_ih_l{}".format(i)
+                attr_weight_hh_li = "weight_hh_l{}".format(i)
+                weight_ih = "weight_ih".format(i)
+                weight_hh = "weight_hh".format(i)
+
+                weight_ih_li_pytorch = getattr(gru_pytorch.brain, attr_weight_ih_li).detach().numpy()
+                weight_hh_li_pytorch = getattr(gru_pytorch.brain, attr_weight_hh_li).detach().numpy()
+
+                weight_ih_li_numpy = getattr(gru_numpy, weight_ih)[i].reshape(weight_ih_li_pytorch.shape)
+                weight_hh_li_numpy = getattr(gru_numpy, weight_hh)[i].reshape(weight_hh_li_pytorch.shape)
+
+                assert np.array_equal(weight_ih_li_pytorch, weight_ih_li_numpy)
+                assert np.array_equal(weight_hh_li_pytorch, weight_hh_li_numpy)
+
+                # Check hidden and cell states
+                assert np.array_equal(gru_pytorch.hidden[i].view(-1).detach().numpy(), gru_numpy.hidden[i][0])
+
```

Comment:
Die Matrizen für die Ausgabegewichte werden glaube ich nicht überprüft, oder? Bei Elman und LSTM auch so

```suggestion

        assert np.array_equal(gru_pytorch.weight_ho, gru_numpy.weight_ho)

```